### ISSUE TITLE
[Feature Request]: Support Response::macro. 

### ISSUE THREAD

**Issue by chrisZng** [laravel-idea/plugin#640]
### Feature Description

https://laravel.com/docs/9.x/responses#response-macros

> Method 'macro' not found in \Illuminate\Support\Facades\Response

```
<?php
 
namespace App\Providers;
 
use Illuminate\Support\Facades\Response;
use Illuminate\Support\ServiceProvider;
 
class AppServiceProvider extends ServiceProvider
{
    /**
     * Bootstrap any application services.
     *
     * @return void
     */
    public function boot()
    {
        Response::macro('caps', function ($value) {
            return Response::make(strtoupper($value));
        });
    }
}
```

> Method 'caps' not found in \Illuminate\Contracts\Foundation\Application|\Illuminate\Contracts\Routing\ResponseFactory|\Illuminate\Http\Response 
 Inspection info: Reports the references to class methods that are not defined.

> Cannot find declaration to go to

```
return response()->caps('foo');
```

PhpStorm: 2022.2.2
Laravel Idea: 5.8.2.222
menu op: Laravel -> Generate Helper Code.

**Comment by adelf** [comment]
Yes. You're adding it to the facade... If you add it to the corresponding class `\Illuminate\Routing\ResponseFactory` - it will work both in this class and in the facade.

**Comment by adelf** [comment]
BTW, don't close it, I'll try to make it work even in this case.

**Comment by JorgenSolli** [comment]
Do you happen to have any news on this @adelf?

**Comment by mirko77** [comment]
Same here

```
Response::macro('attachment', function ($content, $filename) {
            $headers = [
                'Content-type' => 'text/csv',
                'Content-Disposition' => 'attachment; filename="' . $filename . '"',
            ];
            return Response::make($content, 200, $headers);
        });
 ```
 
 `return Response::attachment($content, $filename);`
 
 `Method 'attachment' not found in \Illuminate\Support\Facades\Response`

**Comment by kerwin-wildsea** [comment]
> Yes. You're adding it to the facade... If you add it to the corresponding class `\Illuminate\Routing\ResponseFactory` - it will work both in this class and in the facade.

@adelf This didn't seem to work for me. The only way for it to be picked up by `response()->...` was to use the contract like so:


**⚠️ DO NOT COPY- THIS DOES NOT WORK ⚠️**

```php
\Illuminate\Contracts\Routing\ResponseFactory::macro(...);
```

The problem is that macro cannot be called on the contract. 

Is there another way around this? Every way to make it function in PHP is not picked up by Laravel Idea, and if it is picked up by Laravel Idea, it's not valid PHP.

**Comment by kerwin-wildsea** [comment]
@adelf Hey, has there been any progress on this? Or is there any way someone can help with this?
I just ran into the same issue but for the `auth()` helper.

Adding the line below makes it work for the plugin with autocomplete etc. 
```php
\Illuminate\Auth\SessionGuard::macro(...);
```
But this doesn't actually work. It throws a PHP error. You actually have to use:
```php
\Illuminate\Auth\RequestGuard::macro(...);
```
But this does not come up in auto completion. 

Edit:
Workaround for now is adding both, one for autocomplete, one for functionality:
```php
SessionGuard::macro('key', fn (): int => null);  /* Allow autocomplete */
RequestGuard::macro('key', function (): int {
    // ...
});
```

**Comment by gavinliu6** [comment]
Is there any progress? @adelf

**Comment by adelf** [comment]
It was fixed in Laravel Idea 10.

**Comment by kerwin-wildsea** [comment]
> @adelf It was fixed in Laravel Idea 10.

Thank you! 

Just a quick note, the [changelog](https://laravel-idea.com/docs/changelog#version-10-0-2024-02-13) states the wrong year for the release: `Version 10.0 (2024-02-13)`

**Comment by adelf** [comment]
Fixed. Thanks.

**Comment by gavinliu6** [comment]
```php
Response::macro('success', function ($data = [], int $status = 200, array $headers = []) {
    // do sth.
});

response()->success(); // not working
```

**Comment by mirko77** [comment]
@gavinliu6 try

```
Response::success()

```

**Comment by gavinliu6** [comment]
`\Illuminate\Support\Facades\Response::success()` does work, but wouldn't `response()->success()` also work?

**Comment by gorbunov** [comment]
> `\Illuminate\Support\Facades\Response::success()` does work, but wouldn't `response()->success()` also work?

I'll reopen issue for time being, I don't see much problem implementing it right now, will take a look

**Comment by adelf** [comment]
Fixed in 11.4.
